custom_components/tahoma/sensor.py

```diff
@@ -104,35 +115,9 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
-        attribute = [
-            attr.value
-            for attr in self.device.attributes
-            if attr.name == CORE_MEASURED_VALUE_TYPE
-        ]
-        TEMPERATURE_UNIT = {
-            CORE_TEMPERATURE_IN_CELSIUS: TEMP_CELSIUS,
-            CORE_TEMPERATURE_IN_CELCIUS: TEMP_CELSIUS,
-            CORE_TEMPERATURE_IN_KELVIN: TEMP_KELVIN,
-            CORE_TEMPERATURE_IN_FAHRENHEIT: TEMP_FAHRENHEIT,
-        }.get(attribute[0], TEMP_CELSIUS,)
-        state = self.select_state(
-            CORE_CO2_CONCENTRATION_STATE,
-            CORE_CO_CONCENTRATION_STATE,
-            CORE_ELECTRIC_ENERGY_CONSUMPTION_STATE,
-            CORE_ELECTRIC_POWER_CONSUMPTION_STATE,
-            CORE_LUMINANCE_STATE,
-            CORE_RELATIVE_HUMIDITY_STATE,
-            CORE_TEMPERATURE_STATE,
-        )
-        return {
-            CORE_TEMPERATURE_STATE: TEMPERATURE_UNIT,
-            CORE_RELATIVE_HUMIDITY_STATE: UNIT_PERCENTAGE,
-            CORE_LUMINANCE_STATE: UNIT_LX,
-            CORE_ELECTRIC_POWER_CONSUMPTION_STATE: POWER_WATT,
-            CORE_ELECTRIC_ENERGY_CONSUMPTION_STATE: ENERGY_WATT_HOUR,
-            CORE_CO_CONCENTRATION_STATE: CONCENTRATION_PARTS_PER_MILLION,
-            CORE_CO2_CONCENTRATION_STATE: CONCENTRATION_PARTS_PER_MILLION,
-        }.get(state)
+        attribute = self.device.attributes[CORE_MEASURED_VALUE_TYPE]
+        return UNITS.get(attribute.value) if attribute else None
```

Comment:
isn't `.get(attribute.value)` already returning None as a default?

Reply:
You are right, I will simply this.

Reply:
my bad, won't work, `attribute.value` won't work if `attribute` is `None` cause `NoneType` has no attribute `value`

Reply:
Sorry, it was correct in fact. I have a sensor ('io:EnergyConsumptionSensorsHeatPumpComponent') which has no attributes field. I set this condition for such case.